### PR TITLE
SubModel fundamentally broken, fix it!

### PR DESCRIPTION
**Pull request by nirvn**
There's no saying how many obscure bugs this might have caused.

### CODE REVIEW

File: src/qml/editorwidgets/TextEdit.qml [hunk 1]
```diff
@@ -71,7 +71,7 @@ EditorWidgetBase {
     background: Rectangle {
         width: parent.width
         height: parent.height
-        color: white;
+        color: "white";
```
Comment (m-kuhn) [inline src/qml/editorwidgets/TextEdit.qml:74]: And a goodie ;-)

**Review by m-kuhn — APPROVED** [review]
Thanks.

I guess it's safe to assume if 'first' is in model, everything is in model

**Comment by m-kuhn** [comment]
> Thanks.
> 
> I guess it's safe to assume if 'first' is in model, everything is in model

Actually should we qassert that?

**Comment by nirvn** [comment]
We do attach root of model to submodel,  we need to support that. Which this does too (like it did in the past).

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6304e2d75f2830c2aea463713945d6987eade1ae:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-93b0120f4b9d534c32cfec156a11f98ade77dc81-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-93b0120f4b9d534c32cfec156a11f98ade77dc81-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-93b0120f4b9d534c32cfec156a11f98ade77dc81-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-93b0120f4b9d534c32cfec156a11f98ade77dc81-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 59f122938d027f0cac1f1eddf5f7c7c6c59cb1ff:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-3045bc1a5898aa453bc8dca03de7f1dc04ea8e9b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-3045bc1a5898aa453bc8dca03de7f1dc04ea8e9b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-3045bc1a5898aa453bc8dca03de7f1dc04ea8e9b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-3045bc1a5898aa453bc8dca03de7f1dc04ea8e9b-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c2f9a81094932a58860ceeb59c08eab60e8a1344:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-db7e3bd822b25cead9e4b0f0c63c8c723a4e29ac-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-db7e3bd822b25cead9e4b0f0c63c8c723a4e29ac-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-db7e3bd822b25cead9e4b0f0c63c8c723a4e29ac-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1748-db7e3bd822b25cead9e4b0f0c63c8c723a4e29ac-x86.apk)

**Pull request merged**